### ISSUE TITLE
Use better syntax for specifying dependency versions

### ISSUE THREAD

**Issue by purva-thakre** [vprusso/toqito#933]
The dependencies mentioned in `pyproject.toml` create dependabot PRs on a regular basis. 

Instead of using `=` we want to use other specifiers such that the daily dependabot PRs are reduced.

**Comment by vprusso** [comment]
What other specifiers would you suggest be preferably to reduce the dependabot noise?

Alternatively, we could also consider the cadence of the dependabot to be not daily, but say, weekly as well.

**Comment by purva-thakre** [comment]
I am not too sure. From what I recently figured out, `==` and `=` lead to excessive dependabot PRs. 

> Alternatively, we could also consider the cadence of the dependabot to be not daily, but say, weekly as well.

We could try this out.

**Comment by purva-thakre** [comment]
@vprusso What do you think about the newer frequency of dependabot PRs? 

We could mess around with what updates we allow etc. https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#dependency-type-allow

**Comment by purva-thakre** [comment]
Something is definitely wrong with the syntax or dependabot settings. 

dependabot has created duplicated PRs. I have already merged #976 but #984 was created for the same version upgrade.

**Comment by purva-thakre** [comment]
@vprusso In #987 I changed dependabot settings to only allow major version upgrades. Let me know if you want to go back to the previous settings.

I will keep this issue open for the next month to observe how dependabot behaves.

**Comment by vprusso** [comment]
@purva-thakre any thoughts as to whether dependabot is still misbehaving here? I think things have been functioning well for the past few weeks (unless maybe I'm incorrect?). If things are going well though, perhaps we can close this issue out?

**Comment by purva-thakre** [comment]
No, I don't think dependabot is misbehaving in any way. Looks like the 'direct' updates worked.
